docs/source/api/consistenthash.rst

```diff
@@ -0,0 +1,50 @@
+..
+..
+.. Licensed under the Apache License, Version 2.0 (the "License");
+.. you may not use this file except in compliance with the License.
+.. You may obtain a copy of the License at
+..
+..     http://www.apache.org/licenses/LICENSE-2.0
+..
+.. Unless required by applicable law or agreed to in writing, software
+.. distributed under the License is distributed on an "AS IS" BASIS,
+.. WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+.. See the License for the specific language governing permissions and
+.. limitations under the License.
+..
+
+.. _to-api-consistenthash:
+
+***************
+``consistenthash``
+***************
+Test pattern based consistent hashing for a Delivery Service using a regex and a request path
+
+``POST``
```

Comment:
whenever i think of `POST` i think of create. i guess it's kind of "creating" a consistent hash but not really in the sense that other POSTs create. What about something like:

GET /api/1.4/consistenthash?regex=&requestpath=&cndId=

but not sure if a regex value would make that impossible

Reply:
Yeah, the regex can contain special characters. It'd have to be encoded, which would be awkward and difficult to use.

I'd vote keeping it a POST. It feels odd, but I think it's a pretty standard technique for APIs, for endpoints that require data with special characters or large amounts of data.

Reply:
I started with a `GET` but I had issues encoding the URI components so I went to a `POST`. It's not a typical `GET` either, but I can try to change it if that is the consensus.

Reply:
I actually like POST for this - associating POST with object creation is a pitfall of working with the TO API imo. What it really means is "process this data", and what this does is ask the server to perform operations on the data, and not change server state in any way. The results of the request are the results of those operations, not just data stored on the server. POST seems appropriate.